packages/grafana-data/src/transformations/transformers/merge/dataFrameBuilder.ts

```diff
@@ -0,0 +1,131 @@
+import { MutableDataFrame } from '../../../dataframe';
+import { DataFrame, FieldType, Field } from '../../../types/dataFrame';
+import { ArrayVector } from '../../../vector';
+import { omit } from 'lodash';
+import { getFrameDisplayName } from '../../../field';
+
+interface DataFrameBuilderResult {
+  dataFrame: MutableDataFrame;
+  valueMapper: ValueMapper;
+}
+
+type ValueMapper = (frame: DataFrame, valueIndex: number, timeIndex: number) => Record<string, any>;
+
+const fieldNames = {
+  metric: 'Metric',
+  time: 'Time',
+  value: 'Value',
+};
+
+export class DataFrameBuilder {
```

Comment:
Nit: name of this file should be `DataFrameBuilder` according to https://github.com/grafana/grafana/blob/master/contribute/style-guides/frontend.md#file-and-directory-naming-conventions